### PR TITLE
8344443: Deprecate FXPermission for removal

### PR DESCRIPTION
**Pull request by kevinrushforth**
This PR deprecates the FXPermission class for removal. The intent is to deprecate it in 24 and remove it in 26.

This class no longer does anything. Now that we have finished the post-SM removal cleanup, there are no remaining references FXPermission in JavaFX.

/csr
/reviewers 2

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires CSR request [JDK-8344456](https://bugs.openjdk.org/browse/JDK-8344456) to be approved
- [x] Change must be properly reviewed (2 reviews required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer), 1 [Author](https://openjdk.org/bylaws#author))

### Issues
 * [JDK-8344443](https://bugs.openjdk.org/browse/JDK-8344443): Deprecate FXPermission for removal (**Enhancement** - P4)
 * [JDK-8344456](https://bugs.openjdk.org/browse/JDK-8344456): Deprecate FXPermission for removal (**CSR**)

### Reviewers
 * [Andy Goryachev](https://openjdk.org/census#angorya) (@andy-goryachev-oracle - **Reviewer**)
 * [Ambarish Rapte](https://openjdk.org/census#arapte) (@arapte - **Reviewer**)
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jfx.git pull/1643/head:pull/1643` \
`$ git checkout pull/1643`

Update a local copy of the PR: \
`$ git checkout pull/1643` \
`$ git pull https://git.openjdk.org/jfx.git pull/1643/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1643`

View PR using the GUI difftool: \
`$ git pr show -t 1643`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jfx/pull/1643.diff">https://git.openjdk.org/jfx/pull/1643.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jfx/pull/1643#issuecomment-2483880447)
</details>

### CODE REVIEW

File: modules/javafx.base/src/main/java/javafx/util/FXPermission.java [hunk 1]
```diff
@@ -43,7 +43,12 @@
  * @see java.security.PermissionCollection
  *
  * @since 9
+ *
+ * @deprecated This class was only useful in connection with the
+ *       Security Manager, which is no longer supported.
```
Comment (andy-goryachev-oracle) [inline modules/javafx.base/src/main/java/javafx/util/FXPermission.java:48]: should it be SecurityManager (as in class name)?

Comment (kevinrushforth) [reply modules/javafx.base/src/main/java/javafx/util/FXPermission.java:48]: No, this refers to "the Security Manager" as a feature, which is more than the `SecurityManager` class. See https://openjdk.org/jeps/486.

**Review by andy-goryachev-oracle — APPROVED** [review]

**Comment by kevinrushforth** [comment]
@arapte Can you be the second reviewer?

@prrace You might want to take a look as well, since you are planning to do the same thing for `AWTPermission` at some point.

**Review by arapte — APPROVED** [review]
LGTM, FXPermission is not used anywhere.

**Review by prrace — APPROVED** [review]

**Pull request closed without merging**